plenum/server/consensus/ordering_service.py

```diff
@@ -2332,3 +2339,63 @@ def get_sent_commit(self, viewNo, ppSeqNo):
 
     def replica_batch_digest(self, reqs):
         return replica_batch_digest(reqs)
+
+    def process_view_change_started(self, msg: ViewChangeStarted):
+        # 1. update shared data
+        self._data.preprepared = []
+        self._data.prepared = []
+
+        # 2. save existing PrePrepares
+        new_old_view_preprepares = {(pp.ppSeqNo, pp.digest): pp
+                                    for pp in itertools.chain(self.prePrepares.values(), self.sentPrePrepares.values())}
+        self.old_view_preprepares.update(new_old_view_preprepares)
+
+        # 3. revert unordered transactions
+        if self.is_master:
+            self.revert_unordered_batches()
+
+        # 4. Clear the 3PC log
+        self.prePrepares.clear()
+        self.prepares.clear()
+        self.commits.clear()
+
+        self.requested_pre_prepares.clear()
+        self.requested_prepares.clear()
+        self.requested_commits.clear()
+
+        self.pre_prepare_tss.clear()
+        self.prePreparesPendingFinReqs.clear()
+        self.prePreparesPendingPrevPP.clear()
+        self.sentPrePrepares.clear()
+        self.batches.clear()
+        self.ordered.clear_below_view(msg.view_no)
+        return PROCESS, None
+
+    def process_new_view_checkpoints_applied(self, msg: NewViewCheckpointsApplied):
```

Comment:
This method is WIP (just sufficient to make the integration test pass). It will be finalized in the scope of lNDY-1340